### PR TITLE
Finish session "Explaining the Internet"

### PR DESCRIPTION
**Pull request by teodutu**
- [x] Finish session content
- [ ] Add challenges

### CODE REVIEW

File: explaining-the-internet/index.md [hunk 1]
```diff
@@ -106,8 +107,10 @@ This connection can be:
 This layer is also known as the **Data Link** layer.
 
 It represents the underlying technology of any application.
-The device on which the application is running may have a choice of many technologies to connect to a network, such as Ethernet, WiFi, Bluetooth, 4G, 5G etc.
-Where more than one link exists, the operating system of the device chooses the most appropriate link, e.g. a mobile phone might be connected to both 4G and WiFi.
+The device on which the application is running may have a choice of many technologies to connect to a network, such as Ethernet, WiFi, Bluetooth, 4G, 5G (_tinfoil hat off_) etc.
+Where more than one link exists, the operating system of the device chooses the most appropriate link.
+For instance, a mobile phone might be connected to both 4G and WiFi.
+Most moblile phones prefer the WiFi connection, but remain connected to 4G mobile data, which they use as backup.
```
Comment (gabrielmocanu) [inline explaining-the-internet/index.md:113]:
```suggestion
Most mobile phones prefer the WiFi connection, but remain connected to 4G mobile data, which they use as backup. 
```

File: explaining-the-internet/index.md [hunk 2]
```diff
@@ -135,7 +138,11 @@ You need some intermediaries: **routers**.
 
 ### Routers
 
-A router is a networking device that connects two or more networks.
+It is impossible to connect your PC / laptop directly (physically) to every PC or server in the world.
+Therefore, we need **networks**, which are basically aggregations of hosts (servers, PCs, laptops, mobile phones etc.).
+
+Each network contains one **router** which sends data between the hosts in its network and those on other netowrks.
```
Comment (gabrielmocanu) [inline explaining-the-internet/index.md:144]:
```suggestion
Each network contains one **router** which sends data between the hosts in its network and those on other networks.
```

File: explaining-the-internet/index.md [hunk 3]
```diff
@@ -247,6 +262,8 @@ An unreachable host is as good as a dead one: _useless_.
 `localhost`, in other words. the `lo` interface, should **always** be up and respond to pings.
 In case it doesn't, well... your kernel's TCP/IP stack may be broken.
 Sad reacts only.
```
Comment (gabrielmocanu) [inline explaining-the-internet/index.md:264]: I don't think this fits well here.
```suggestion
```

File: explaining-the-internet/index.md [hunk 4]
```diff
@@ -396,13 +413,29 @@ Notice that only the domain of a URL gets DNS'd.
...
+security.summer.schoo.github.io has IPv6 address 2606:50c0:8000::153
+security.summer.schoo.github.io has IPv6 address 2606:50c0:8003::153
+security.summer.schoo.github.io has IPv6 address 2606:50c0:8001::153
+```
 
-TODO
+Notice there are more IP's for `security.summer.schoo.github.io`.
+The main reason for this is **load balancing**.
+Github is accessed frequently and in order not to overload a single server, it uses multiple servers for a single domain.
+Obviously, each of them serve the same web page: `security.summer.schoo.github.io`.
```
Comment (gabrielmocanu) [inline explaining-the-internet/index.md:437]:
```suggestion
Obviously, each of them serves the same web page: `security.summer.schoo.github.io`.
```

**Comment by gabrielmocanu** [comment]
Any reasons to drop information about Nmap on the Port section?

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

**Comment by teodutu** [comment]
> Any reasons to drop information about Nmap on the Port section?

Yes, `nmap` needs a discussion about network masks, which would take a very long time from an already very theoretically-heavy session. It kind of sucks now because it's left me without one challenge idea, but it might be for the better overall.

File: explaining-the-internet/index.md (hunk 3 above)
Comment (teodutu) [reply explaining-the-internet/index.md:264]: You're such a boomer. But _sometimes_ boomers are right.

**Comment by teodutu** [comment]
>  Notice that only the domain of a URL gets DNS'd. -> I don't get it.. DNS'd ??

`get DNS'd` = have its name resolved to an IP address by the DNS

But I get it, I won't invent words anymore.

**Comment by teodutu** [comment]
@gabrielmocanu I've added your suggestions. Please have another look. I discarded those with spelling suggestions such as:
- `*ise` -> `*ize`
- `favourite` -> `favorite`

All spelling are correct. I just prefer the former ones (British English) as opposed to the latter (American English).

In addition, do you have any ideas for another 1 or 2 challenges?

**Comment by gabrielmocanu** [comment]
Use john to brute-force password when login in with ssh.

**Pull request merged**